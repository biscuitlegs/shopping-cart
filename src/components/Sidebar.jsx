import React from 'react'
import placeholderImage from '../assets/logo192.png'

const Sidebar = () => (
  <div
    className="d-flex flex-column flex-shrink-0 p-3 bg-light"
    data-testid="sidebar"
  >
    <h5 className="fs-4 text-center">Sidebar</h5>
    <hr />
    <ul className="nav nav-pills flex-column mb-auto">
      <li className="nav-item">
        <a href="#" className="nav-link active" aria-current="page">
          <svg className="bi pe-none me-2" width="16" height="16">
            <use xlinkHref="#home" />
          </svg>
          Nulla facilisis
        </a>
      </li>
      <li>
        <a href="#" className="nav-link link-dark">
          <svg className="bi pe-none me-2" width="16" height="16">
            <use xlinkHref="#speedometer2" />
          </svg>
          Etiam elementum
        </a>
      </li>
      <li>
        <a href="#" className="nav-link link-dark">
          <svg className="bi pe-none me-2" width="16" height="16">
            <use xlinkHref="#table" />
          </svg>
          Cras consequat
        </a>
      </li>
      <li>
        <a href="#" className="nav-link link-dark">
          <svg className="bi pe-none me-2" width="16" height="16">
            <use xlinkHref="#grid" />
          </svg>
          Lacinia tempor
        </a>
      </li>
      <li>
        <a href="#" className="nav-link link-dark">
          <svg className="bi pe-none me-2" width="16" height="16">
            <use xlinkHref="#people-circle" />
          </svg>
          Cras tincidunt
        </a>
      </li>
    </ul>
    <hr />
    <ul className="nav nav-pills flex-column mb-auto">
      <li className="nav-item">
        <a href="#" className="nav-link link-dark">
          <svg className="bi pe-none me-2" width="16" height="16">
            <use xlinkHref="#home" />
          </svg>
          Etiam pulvinar
        </a>
      </li>
      <li>
        <a href="#" className="nav-link link-dark">
          <svg className="bi pe-none me-2" width="16" height="16">
            <use xlinkHref="#speedometer2" />
          </svg>
          Integer viverra
        </a>
      </li>
      <li>
        <a href="#" className="nav-link link-dark">
          <svg className="bi pe-none me-2" width="16" height="16">
            <use xlinkHref="#table" />
          </svg>
          Nulla
        </a>
      </li>
      <li>
        <a href="#" className="nav-link link-dark">
          <svg className="bi pe-none me-2" width="16" height="16">
            <use xlinkHref="#grid" />
          </svg>
          Cras sed
        </a>
      </li>
      <li>
        <a href="#" className="nav-link link-dark">
          <svg className="bi pe-none me-2" width="16" height="16">
            <use xlinkHref="#people-circle" />
          </svg>
          Praesent sit
        </a>
      </li>
    </ul>
    <hr />
    <div className="dropdown">
      <a
        href="#"
        className="d-flex align-items-center link-dark text-decoration-none dropdown-toggle"
        id="dropdownUser2"
        data-bs-toggle="dropdown"
        aria-expanded="false"
      >
        <img
          src={placeholderImage}
          alt=""
          width="32"
          height="32"
          className="rounded-circle me-2"
        />
        <strong>Username</strong>
      </a>
      <ul
        className="dropdown-menu text-small shadow"
        aria-labelledby="dropdownUser2"
      >
        <li>
          <a className="dropdown-item" href="#">
            Settings
          </a>
        </li>
        <li>
          <a className="dropdown-item" href="#">
            Profile
          </a>
        </li>
        <li>
          <hr className="dropdown-divider" />
        </li>
        <li>
          <a className="dropdown-item" href="#">
            Sign out
          </a>
        </li>
      </ul>
    </div>
  </div>
)

export default Sidebar
