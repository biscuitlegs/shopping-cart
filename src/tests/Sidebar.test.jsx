import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Sidebar from '../components/Sidebar.jsx'

test('shows a heading', () => {
  render(<Sidebar />)

  const heading = screen.getByRole('heading')

  expect(heading).toBeInTheDocument()
})

test('shows links', () => {
  render(<Sidebar />)

  const links = screen.getAllByRole('link')

  expect(links).toHaveLength(14)
})

test('renders sidebar correctly', () => {
  const tree = renderer.create(<Sidebar />).toJSON()

  expect(tree).toMatchSnapshot()
})
