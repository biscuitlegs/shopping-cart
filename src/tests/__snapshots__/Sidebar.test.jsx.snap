// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sidebar correctly 1`] = `
<div
  className="d-flex flex-column flex-shrink-0 p-3 bg-light"
  data-testid="sidebar"
>
  <h5
    className="fs-4 text-center"
  >
    Sidebar
  </h5>
  <hr />
  <ul
    className="nav nav-pills flex-column mb-auto"
  >
    <li
      className="nav-item"
    >
      <a
        aria-current="page"
        className="nav-link active"
        href="#"
      >
        <svg
          className="bi pe-none me-2"
          height="16"
          width="16"
        >
          <use
            xlinkHref="#home"
          />
        </svg>
        Nulla facilisis
      </a>
    </li>
    <li>
      <a
        className="nav-link link-dark"
        href="#"
      >
        <svg
          className="bi pe-none me-2"
          height="16"
          width="16"
        >
          <use
            xlinkHref="#speedometer2"
          />
        </svg>
        Etiam elementum
      </a>
    </li>
    <li>
      <a
        className="nav-link link-dark"
        href="#"
      >
        <svg
          className="bi pe-none me-2"
          height="16"
          width="16"
        >
          <use
            xlinkHref="#table"
          />
        </svg>
        Cras consequat
      </a>
    </li>
    <li>
      <a
        className="nav-link link-dark"
        href="#"
      >
        <svg
          className="bi pe-none me-2"
          height="16"
          width="16"
        >
          <use
            xlinkHref="#grid"
          />
        </svg>
        Lacinia tempor
      </a>
    </li>
    <li>
      <a
        className="nav-link link-dark"
        href="#"
      >
        <svg
          className="bi pe-none me-2"
          height="16"
          width="16"
        >
          <use
            xlinkHref="#people-circle"
          />
        </svg>
        Cras tincidunt
      </a>
    </li>
  </ul>
  <hr />
  <ul
    className="nav nav-pills flex-column mb-auto"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link link-dark"
        href="#"
      >
        <svg
          className="bi pe-none me-2"
          height="16"
          width="16"
        >
          <use
            xlinkHref="#home"
          />
        </svg>
        Etiam pulvinar
      </a>
    </li>
    <li>
      <a
        className="nav-link link-dark"
        href="#"
      >
        <svg
          className="bi pe-none me-2"
          height="16"
          width="16"
        >
          <use
            xlinkHref="#speedometer2"
          />
        </svg>
        Integer viverra
      </a>
    </li>
    <li>
      <a
        className="nav-link link-dark"
        href="#"
      >
        <svg
          className="bi pe-none me-2"
          height="16"
          width="16"
        >
          <use
            xlinkHref="#table"
          />
        </svg>
        Nulla
      </a>
    </li>
    <li>
      <a
        className="nav-link link-dark"
        href="#"
      >
        <svg
          className="bi pe-none me-2"
          height="16"
          width="16"
        >
          <use
            xlinkHref="#grid"
          />
        </svg>
        Cras sed
      </a>
    </li>
    <li>
      <a
        className="nav-link link-dark"
        href="#"
      >
        <svg
          className="bi pe-none me-2"
          height="16"
          width="16"
        >
          <use
            xlinkHref="#people-circle"
          />
        </svg>
        Praesent sit
      </a>
    </li>
  </ul>
  <hr />
  <div
    className="dropdown"
  >
    <a
      aria-expanded="false"
      className="d-flex align-items-center link-dark text-decoration-none dropdown-toggle"
      data-bs-toggle="dropdown"
      href="#"
      id="dropdownUser2"
    >
      <img
        alt=""
        className="rounded-circle me-2"
        height="32"
        src="logo192.png"
        width="32"
      />
      <strong>
        Username
      </strong>
    </a>
    <ul
      aria-labelledby="dropdownUser2"
      className="dropdown-menu text-small shadow"
    >
      <li>
        <a
          className="dropdown-item"
          href="#"
        >
          Settings
        </a>
      </li>
      <li>
        <a
          className="dropdown-item"
          href="#"
        >
          Profile
        </a>
      </li>
      <li>
        <hr
          className="dropdown-divider"
        />
      </li>
      <li>
        <a
          className="dropdown-item"
          href="#"
        >
          Sign out
        </a>
      </li>
    </ul>
  </div>
</div>
`;
